pages/register.tsx

```diff
@@ -13,6 +13,9 @@ import { formSchema } from '~/shared/zodSchemas';
 import { Label } from '~/components/Label';
 import { handleErrorToast } from '~/utils/handleToasts';
 import { ErrorParagraph } from '~/components/ErrorParagraph';
+import { useMutation } from '@tanstack/react-query';
+import { QueryKeys } from '~/shared/types';
+import { useShowErrorToast } from '~/hooks/useShowErrorToast';
 
```

Comment:
Ehdotan että otetaan formin tyyppi skeemasta, niin on määritelty yhdessä paikassa:
```suggestion
import { z } from 'zod';

type FormData = z.infer<typeof formSchema>;
```

Tämän jälkeen voi tyhjän formin muuttujalle antaa tyypin seuraavasti:
```ts
const EMPTY_FORM_DATA: FormData = {
```
Etuna tässä on se, että jos formiin joskus lisää jotain, niin kun sen lisää skeemaan niin tulee automaattisesti virhe jos uusi kenttä puuttuu `EMPTY_FORM_DATA` muuttujasta.

Reply:
Totta! Tuo on hyvä idea, kiitos! :+1: